application/controllers/SurveyAdministrationController.php

```diff
@@ -3195,7 +3195,145 @@ protected function tabPanelIntegration($survey, $sLang = null)
         ];
         $aData['questions'] = $aQuestions;
 
+        $model = new SurveyURLParameter('search');
+        $model->sid = $survey->sid;
+        $model->searched_value = Yii::app()->request->getParam('search_query');
+
+        $aData['updateUrl'] = Yii::app()->createUrl('surveyAdministration/rendersidemenulink', ['surveyid' => $survey->sid, 'subaction' => 'panelintegration']);
+
+        /*if (isset($_GET['QuestionGroup'])) {
+            $model->attributes = $_GET['QuestionGroup'];
+        }*/
+
+        if (isset($_GET['pageSize'])) {
+            Yii::app()->user->setState('pageSize', (int) $_GET['pageSize']);
+        }
+
+        $aData['model'] = $model;
+
         App()->getClientScript()->registerPackage('jquery-datatable');
         return $aData;
     }
+
+    /**
+     * Method to save URL Params (Panel Integration)
+     *
+     * @throws CException
+     */
+    public function actionSaveUrlParam()
```

Comment:
Can you just explain quickly why a bug fix would require several new controller methods? :thinking:

Reply:
You asked me to remove Datatables and use a reglar grid.

Reply:
My opinion : the explain was : «Panel Integration was not set in 5.X version, never work because no action is done… then need to construct all action (and function)»

Reply:
> You asked me to remove Datatables and use a reglar grid.

Ha! I remember that, but can't find any dev ticket about it. Sloppy of me. Sorry.

Reply:
Oh, we just did on this one :) Maybe I was Sloppy